viewing_party/party.py

```diff
@@ -1,23 +1,155 @@
 # ------------- WAVE 1 --------------------
 
+
 def create_movie(title, genre, rating):
-    pass
+    if not title or not genre or not rating:
+        return None
+    return {"title": title, "genre": genre, "rating": rating}
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+    watchlist = user_data["watchlist"]
+    new_watchlist = []
+
+    for i in range(len(watchlist)):
+        if title == watchlist[i]["title"]:
+            add_to_watched(user_data, watchlist[i])
+        else:
+            new_watchlist.append(watchlist[i])
+    user_data["watchlist"] = new_watchlist
+
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
 
+def get_watched_avg_rating(user_data):
+    if not user_data["watched"]:
+        return 0
+
+    ratings_sum = sum([movie["rating"] for movie in user_data["watched"]])

```

Comment:
Nice use of `sum()` to tally up the total ratings. How would you do something similar if a function like `sum()` didn't exist? Now is a great time to think through how library functions solve these problems!